File: api/src/main/java/kimosabe/api/repository/RoleRepository.java
package kimosabe.api.repository;

import kimosabe.api.model.Role;
import kimosabe.api.model.RoleName;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface RoleRepository extends JpaRepository<Role, Long> {
    public Optional<Role> findByName(RoleName roleName);
}

File: api/src/main/kotlin/com/kimosabe/api/repository/RoleRepository.kt
package com.kimosabe.api.repository

import com.kimosabe.api.model.Role
import com.kimosabe.api.model.RoleName
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface RoleRepository: JpaRepository<Role, Long> {
    fun findByName(roleName : RoleName) : Role?
}